python/ray/data/_internal/execution/interfaces/task_context.py

```diff
@@ -15,6 +20,11 @@ class TaskContext:
     # operator.
     task_idx: int
 
+    # If driver memory exceeds this threshold, warn the user. For now, this
+    # only applies to shuffle ops because most other ops are unlikely to use as
+    # much driver memory.
+    warn_on_driver_memory_usage_bytes: Optional[int] = None
```

Comment:
Can we pass this argument through `DataContext` to `ExchangeTaskScheduler.execute(warn_on_driver_memory_usage_bytes=...)` directly?

`TaskContext` ideally should only contain the unique information per each task. Feeling it does not need to go through `TaskContext` in this case.

Reply:
Thanks, moved to ExchangeTaskScheduler.